### DISCUSSION TITLE
Need Dropdown whose value require on code side.

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#11055] [Q&A]
How to create drop down of selection through field or part? on admin side of orchard core I used the content picker is the right way or not? 
If yes?
I do that in that way but instead of I have the name like Industry News in category selection i receive the ContentItemIds see :

![image](https://user-images.githubusercontent.com/5202291/150632141-79b6b8e9-6f3d-4023-a895-036a6c542abe.png)

on admin side:

![image](https://user-images.githubusercontent.com/5202291/150632156-7257fe9e-b695-4b76-8b6a-04d30b8f685f.png)

Please advise I need the name of industry News instead of of ContentItemIds?

**Comment by hyzx86** [comment]
You can refer to the naming conventions in the documentation to replace the Field without breaking existing OC code.

https://docs.orchardcore.net/en/dev/docs/reference/modules/Templates/#parttype__fieldname

**Reply by saleemhaq** [reply]
> You can refer to the naming conventions in the documentation to replace the Field without breaking existing OC code.
> 
> https://docs.orchardcore.net/en/dev/docs/reference/modules/Templates/#parttype__fieldname

can u please share it with me the working sample I tried but unable to achieve. Please help.

**Comment by ns8482e** [comment]
@saleemhaq  can you share your content type definition for IndustryNews?

**Reply by saleemhaq** [reply]
> 

I created the content type like that below:

![conenttype](https://user-images.githubusercontent.com/5202291/150854089-cd2ed9ee-464e-4b14-a697-607cc074aa91.png)

![image](https://user-images.githubusercontent.com/5202291/150854176-7ef0cbd0-6a49-4e78-9780-e788032e69b7.png)

and finally:

![image](https://user-images.githubusercontent.com/5202291/150854275-a4cacbba-12d5-488d-ab75-0c3cc45f9add.png)

**Reply by ns8482e** [reply]
I guess you see id because you are not using title part.

**Reply by saleemhaq** [reply]
> I guess you see id because you are not using title part.

Now i using the title part as well but still face the same issue please help:

![image](https://user-images.githubusercontent.com/5202291/150868820-19dc2dbd-223f-44f3-9889-11c717bff593.png)

![image](https://user-images.githubusercontent.com/5202291/150868913-4a6a416a-f009-453d-b5c3-0b53ee6cbfd2.png)

![image](https://user-images.githubusercontent.com/5202291/150869051-c16bf097-145d-4b68-aac5-d2ef790da0f1.png)

Please suggest on above?

**Reply by saleemhaq** [reply]
> I guess you see id because you are not using title part.

@ns8482e  Can u please check the above things as i shared the whole scenario?

**Reply by hyzx86** [reply]
Does GraphqlApi work in your scenario?
 
![image](https://user-images.githubusercontent.com/15613121/151109296-d6fb1797-6b82-4268-b39d-0e3a8e901e07.png)

**Reply by ns8482e** [reply]
> > I guess you see id because you are not using title part.
> 
> @ns8482e Can u please check the above things as i shared the whole scenario?

Publish your item with updated TitlePart and it will show. If TitlePart is empty in published version then it will show id

**Reply by hyzx86** [reply]
If you plan to use a title that you can type, fill in the title content when editing the content item
If you want titles to be generated automatically, use the following Settings
Update your   `Category` （I guess） TypeDefinition
Edit Title 
![image](https://user-images.githubusercontent.com/15613121/151178874-aa508740-1d46-42df-ba05-3d9e334b1190.png)

**Reply by hyzx86** [reply]
If you want to say that the title does not show up in the saved content item, you need to develop a `NewContentPicker`,
but as I mentioned above, you can use the Graphql API to automatically query the information associated with the content item
Or write a custom SQL query

**Reply by saleemhaq** [reply]
> If you want to say that the title does not show up in the saved content item, you need to develop a `NewContentPicker`, but as I mentioned above, you can use the Graphql API to automatically query the information associated with the content item Or write a custom SQL query

I make the query but how i call i call or set value when creating post and select the category there?

![image](https://user-images.githubusercontent.com/5202291/151609255-2470510c-8edd-43dc-bfc5-4d8d551f1632.png)

Can u please guide.? @hyzx86

**Comment by ns8482e** [comment]
`TitlePart` is required for `DisplayText` of content item - However if you wish, you can configure `TitlePart` not to be visible in editor and have it computed via liquid

![image](https://user-images.githubusercontent.com/23270244/151187360-cb7cc0c9-d272-4d36-b85a-3c4d01e4d207.png)

**Comment by saleemhaq** [comment]
I used the razor syntax?

**Reply by ns8482e** [reply]
for pattern you need to use liquid - see screenshot above

**Reply by ns8482e** [reply]
Changes made in TitlePart Settings will only apply when new item is added or edited existing item - so remember to update all your items :)

**Reply by saleemhaq** [reply]
> Changes made in TitlePart Settings will only apply when new item is added or edited existing item - so remember to update all your items :)

I used the razor syntax same thing is there still :( see below:

![image](https://user-images.githubusercontent.com/5202291/151601815-c117e14a-af76-4842-a408-ddbe3a42433a.png)

![image](https://user-images.githubusercontent.com/5202291/151601877-b7a0e3e8-97de-4c38-b4f1-f36591a96351.png)

I created new post and link like that: 

![Industry Selection](https://user-images.githubusercontent.com/5202291/151602071-11d1b6e1-911e-4646-8550-6f4494e8ed3e.png)

still i got the same contentid instead of title:

![image](https://user-images.githubusercontent.com/5202291/151602128-691738ea-0ac8-4be2-a2f5-17f27f0037af.png)

@ns8482e please help on above what goes wrong with me :(
